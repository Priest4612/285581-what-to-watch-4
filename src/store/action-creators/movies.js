import ActionType from '../action-type';

const moviesLoaded = (movies) => ({
  type: ActionType.Movies.FETCH_MOVIES_SUCCESS,
  payload: movies,
});

const setActiveMovieId = (id) => ({
  type: ActionType.Movies.CHANGE_ACTIVE_MOVIE,
  payload: id,
});

export default {
  moviesLoaded,
  setActiveMovieId,
};
