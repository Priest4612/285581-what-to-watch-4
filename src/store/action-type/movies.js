export default {
  FETCH_MOVIES_FAILURE: `FETCH_MOVIES_FAILURE`,
  FETCH_MOVIES_SUCCESS: `FETCH_MOVIES_SUCCESS`,
  FETCH_MOVIES_REQUEST: `FETCH_MOVIES_REQUEST`,
  CHANGE_ACTIVE_MOVIE: `CHANGE_ACTIVE_MOVIE`,
};
