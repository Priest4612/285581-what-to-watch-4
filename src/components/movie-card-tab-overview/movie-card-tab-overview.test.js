import React from 'react';
import renderer from 'react-test-renderer';

import movies from '../../mocks/test-films-mock.json';

import MovieCardTabReviews from './movie-card-tab-overview.jsx';


it(`Render MovieCardTabReviews`, () => {
  const {rating, scoresCount, director, starring, description} = movies[1];

  const tree = renderer
    .create(<MovieCardTabReviews
      rating={rating}
      scoresCount={scoresCount}
      director={director}
      starring={starring}
      description={description}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
