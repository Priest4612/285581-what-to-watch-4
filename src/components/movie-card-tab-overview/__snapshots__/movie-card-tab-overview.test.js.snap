// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieCardTabReviews 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      6.6
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        901 ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: Я нёс цветы, дед свою поломанную ветку, мы шли молча.
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: Дэвид Линч, Альфред Хичкок, Сидни Люмет, Квентин Тарантино, Майкл Манн, Кристофер Нолан, Леонид Гайдай, Билли Уайлдер
      </strong>
    </p>
  </div>,
]
`;
